roles/build_npm/tasks/main.yml

```diff
@@ -19,14 +19,17 @@
         channel: "{{ node_version }}/stable"
         state: present
       when: ansible_distribution == "Ubuntu"
+      register: snap_results
 
-    # NOTE: upgrading nodejs on cdh-geniza1 failed;
-    # was able to get it working with refresh
-    # instead of install:
-    #   sudo snap refresh node --channel=16
-    # We may want to run refresh as a command, see
-    # https://serverfault.com/a/1025300
+    # NOTE: upgrading nodejs with ansible snap fails, even though
+    # the documentation claims it should refresh when then channel changes.
+    # Manually run a refresh command to ensure version changes take effect, e.g.:
+    #   sudo snap refresh node --channel=18
+    # NOTE2: could add a node -v check and only refresh on mismatch
 
+    - name: Refresh nodejs to ensure version changes take effect
+      become: true
+      ansible.builtin.command: "snap refresh node --channel={{ node_version }}/stable"
```

Comment:
@kayiwa FYI, we (@quadrismegistus and I) needed to solve the node upgrade problem (#143) and came up with a simpler solution than switching to the princeton ansible role (as previously proposed). Running the refresh command here resolves the problem, and doesn't seem to cause any problems or delay when refresh has been done. The comment above should make it clear why this is here, and if/when ansible actually runs the refresh as it claims to do this stanza can be removed.